src/exec/builtins/functions.rs

```diff
@@ -40,6 +40,14 @@ use crate::state::{EnvVariable, Path, ShellState};
 enum DirectoryListPermissionMode {
     Octal,
     String,
+    Hidden,
+}
+
+enum DirectoryListTimestampMode {
```

Comment:
This would be better as a `ValueEnum` instead of multiple `bool` flags in my opinion.